tests/load/dependencies.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import "context"
+
+type Issuer[T comparable] interface {
+	// GenerateAndIssueTx generates and sends a tx to the network, and informs the
+	// tracker that it sent said transaction. It returns the sent transaction.
+	GenerateAndIssueTx(ctx context.Context) (tx T, err error)
+}
+
+type Listener[T comparable] interface {
+	// Listen for the final status of transactions and notify the tracker
+	// Listen stops if the context is done, an error occurs, or if it received
+	// all the transactions issued and the issuer no longer issues any.
+	// Listen MUST return a nil error if the context is canceled.
+	Listen(ctx context.Context) error
+
+	// RegisterIssued informs the listener that a transaction was issued.
+	RegisterIssued(tx T)
+
+	// IssuingDone informs the listener that no more transactions will be issued.
+	IssuingDone()
```

Comment:
I'm wondering if we can simplify this interface. Can't we just have a single function like `Wait(ctx context.Context, tx T) error`? We could just call `Wait` after we issue each transaction, and block on all goroutines running `Wait` when we're doing issuing txs.

Reply:
We don't want to issue->listen for each tx sequentially, we want to issue and listen asynchronously.
Or do you mean to run a goroutine per transaction issuance - if so, I don't think it would scale well and would make code possibly more complex I think. Unless I misunderstood what you mean? 🤔